Analisis en Python 

1. Carga de librerias


In [37]:
import pandas as pd
import matplotlib as plt
import seaborn as sns
import numpy as np 

2. Exportar y explorar los dataframe

Explorando el archivo 1

In [38]:
df_company_trips=pd.read_csv('/datasets/project_sql_result_01.csv')
df_company_trips.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64 entries, 0 to 63
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   company_name  64 non-null     object
 1   trips_amount  64 non-null     int64 
dtypes: int64(1), object(1)
memory usage: 1.1+ KB


In [39]:
df_company_trips.head(10)

company_name  trips_amount
0                          Flash Cab         19558
1          Taxi Affiliation Services         11422
2                  Medallion Leasing         10367
3                         Yellow Cab          9888
4    Taxi Affiliation Service Yellow          9299
5          Chicago Carriage Cab Corp          9181
6                       City Service          8448
7                           Sun Taxi          7701
8          Star North Management LLC          7455
9  Blue Ribbon Taxi Association Inc.          5953

In [40]:
df_company_trips.isna().sum()

company_name    0
trips_amount    0
dtype: int64

In [41]:
df_company_trips.duplicated().sum()

0

Después de realizar la exploración inicial del archivo se revisó si contenia valores null, duplicados y que la información que contiene corresponde al tipo de datos. Este archivo está completo y cumple con los tipos de datos.

Exploarando el archivo 2

In [42]:
dropoff_and_avgtrips=pd.read_csv('/datasets/project_sql_result_04.csv')
dropoff_and_avgtrips.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 94 entries, 0 to 93
Data columns (total 2 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   dropoff_location_name  94 non-null     object 
 1   average_trips          94 non-null     float64
dtypes: float64(1), object(1)
memory usage: 1.6+ KB


In [43]:
dropoff_and_avgtrips.head(10)

dropoff_location_name  average_trips
0                  Loop   10727.466667
1           River North    9523.666667
2         Streeterville    6664.666667
3             West Loop    5163.666667
4                O'Hare    2546.900000
5             Lake View    2420.966667
6            Grant Park    2068.533333
7         Museum Campus    1510.000000
8            Gold Coast    1364.233333
9    Sheffield & DePaul    1259.766667

In [44]:
dropoff_and_avgtrips.isna().sum()

dropoff_location_name    0
average_trips            0
dtype: int64

In [45]:
dropoff_and_avgtrips.duplicated().sum()

0

El archivo 2 no tiene valores faltantes, ni duplicados y los tipos de datos están correctos.

Explorando el archivo 3

In [46]:
weather_records=pd.read_csv('/datasets/project_sql_result_07.csv')
weather_records.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1068 entries, 0 to 1067
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   start_ts            1068 non-null   object 
 1   weather_conditions  1068 non-null   object 
 2   duration_seconds    1068 non-null   float64
dtypes: float64(1), object(2)
memory usage: 25.2+ KB


In [47]:
weather_records.head(10)

start_ts weather_conditions  duration_seconds
0  2017-11-25 16:00:00               Good            2410.0
1  2017-11-25 14:00:00               Good            1920.0
2  2017-11-25 12:00:00               Good            1543.0
3  2017-11-04 10:00:00               Good            2512.0
4  2017-11-11 07:00:00               Good            1440.0
5  2017-11-11 04:00:00               Good            1320.0
6  2017-11-04 16:00:00                Bad            2969.0
7  2017-11-18 11:00:00               Good            2280.0
8  2017-11-11 14:00:00               Good            2460.0
9  2017-11-11 12:00:00               Good            2040.0

In [48]:
weather_records.isna().sum()

start_ts              0
weather_conditions    0
duration_seconds      0
dtype: int64

In [49]:
weather_records.duplicated().sum()

197

Este tercer no presenta valores ausentes, pero el tipo de dato asignado para la columna de fecha no corresponde por lo cual antes de proseguir habrá que corregirlo, también aparecen 197 filas duplicadas, estas se revisaran y posteriormente, si es necesario, se procederá a eliminarlas. 

In [50]:
weather_records['duplicated']=weather_records.duplicated()

In [51]:
weather_records.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1068 entries, 0 to 1067
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   start_ts            1068 non-null   object 
 1   weather_conditions  1068 non-null   object 
 2   duration_seconds    1068 non-null   float64
 3   duplicated          1068 non-null   bool   
dtypes: bool(1), float64(1), object(2)
memory usage: 26.2+ KB


In [52]:
weather_records[weather_records['duplicated']==True].value_counts()


start_ts             weather_conditions  duration_seconds  duplicated
2017-11-11 08:00:00  Good                1380.0            True          7
2017-11-11 06:00:00  Good                1260.0            True          7
                                         1380.0            True          6
2017-11-11 08:00:00  Good                1200.0            True          5
2017-11-11 16:00:00  Good                2400.0            True          5
                                                                        ..
2017-11-04 15:00:00  Good                3180.0            True          1
2017-11-04 14:00:00  Good                3840.0            True          1
                                         3480.0            True          1
                                         3360.0            True          1
2017-11-25 12:00:00  Good                1440.0            True          1
Length: 115, dtype: int64

In [53]:
weather_records[weather_records['start_ts']=='2017-11-11 08:00:00'].value_counts()

start_ts             weather_conditions  duration_seconds  duplicated
2017-11-11 08:00:00  Good                1380.0            True          7
                                         1200.0            True          5
                                         1320.0            True          4
                                         1440.0            True          2
                                         1063.0            False         1
                                         1560.0            False         1
                                         1519.0            False         1
                                         1500.0            True          1
                                                           False         1
                                         1470.0            False         1
                                         1440.0            False         1
                                         1417.0            False         1
                              

Después de revisar los datos duplicados, se decidirá, por el momento, no eliminar los valores duplicados ya que no se sabecomo fueron recolectados y cargados los datos en la base datos,se eliminará la columna creada para estudiar las filas duplicadas

In [56]:
#Arreglando el tipo de dato en la columna start_ts
weather_records['start_ts']=pd.to_datetime(weather_records['start_ts'], format= '%Y-%m-%d %H:%M:%S')

In [58]:
weather_records.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1068 entries, 0 to 1067
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   start_ts            1068 non-null   datetime64[ns]
 1   weather_conditions  1068 non-null   object        
 2   duration_seconds    1068 non-null   float64       
 3   duplicated          1068 non-null   bool          
dtypes: bool(1), datetime64[ns](1), float64(1), object(1)
memory usage: 26.2+ KB


In [59]:
weather_records.drop(columns='duplicated', axis=1)

start_ts weather_conditions  duration_seconds
0    2017-11-25 16:00:00               Good            2410.0
1    2017-11-25 14:00:00               Good            1920.0
2    2017-11-25 12:00:00               Good            1543.0
3    2017-11-04 10:00:00               Good            2512.0
4    2017-11-11 07:00:00               Good            1440.0
...                  ...                ...               ...
1063 2017-11-25 11:00:00               Good               0.0
1064 2017-11-11 10:00:00               Good            1318.0
1065 2017-11-11 13:00:00               Good            2100.0
1066 2017-11-11 08:00:00               Good            1380.0
1067 2017-11-04 16:00:00                Bad            2834.0

[1068 rows x 3 columns]

Buscando los barrios con mayor flujo de viajes